### ISSUE TITLE
"Steamwebhelper is not responding" in latest Steam beta with SDL_VIDEODRIVER=wayland set

### ISSUE THREAD

**Issue by ConnorJC3** [ValveSoftware/steam-for-linux#11358]
#### Your system information

* Steam client version (build number or date): Public beta available on 2024-10-19 (for reasons that will become obvious, I cannot easily get a build number)
* Distribution (e.g. Ubuntu): Arch Linux (Kernel `6.11.4-zen1-1-zen`)
* Opted into Steam client beta?: [Yes/No]: Yes - This bug only occurs on the beta
* Have you checked for system updates?: [Yes/No]: Yes
* Steam Logs: [generate by running this command in a terminal `tar -zcvf ~/Desktop/steam-logs.tar.gz ~/.steam/steam/logs`]: 
[steam-logs.tar.gz](https://github.com/user-attachments/files/17445258/steam-logs.tar.gz)

* GPU: 6900XT using vulkan-radeon (not amdvlk)

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen. Please link any large code pastes as a [Github Gist](https://gist.github.com/)

Attempting to launch steam when using the public beta, it gets stuck and displays an error message that steamwebhelper has crashed:
![image](https://github.com/user-attachments/assets/42394372-8456-48c4-87c6-34e2016e1993)

This issue occurs regardless of launching steam with native libraries or the steam runtime. Reverting to the stable client (i.e. `rm ~/.steam/steam/package/beta`), this issue does not occur. 

#### Steps for reproducing this issue:

1. Close steam
2. Opt into the public beta (`echo publicbeta > ~/.steam/steam/package/beta`)
3. Attempt to launch steam

**Comment by kisak-valve** [comment]
Hello @ConnorJC3, `ERROR: CCEFThread Init failed to initialize SDL: wayland not available` in your webhelper.txt looks like a potential hint.

Can you check if something is setting SDL_VIDEODRIVER=wayland or SDL_VIDEO_DRIVER=wayland in your environment and try running Steam from a terminal with something like `unset SDL_VIDEODRIVER; steam`?

**Comment by Frestein** [comment]
> unset SDL_VIDEODRIVER; steam

Hey! I did that and after that the steam opened successfully.

Arch Linux
Kernel Name: Linux
Kernel Version: 6.11.4-zen1-1-zen
X Server Vendor: The X.Org Foundation
X Server Release: 12401003
X Window Manager: Hyprland :D
Steam Runtime Version: steam-runtime_0.20240916.101793

Client Information:
Version: 1729208001
Browser GPU Acceleration Status: Enabled
Browser Canvas: Enabled
Browser Canvas out-of-process rasterization: Enabled
Browser Direct Rendering Display Compositor: Disabled
Browser Compositing: Enabled
Browser Multiple Raster Threads: Enabled
Browser OpenGL: Enabled
Browser Rasterization: Enabled
Browser Raw Draw: Disabled
Browser Skia Graphite: Disabled
Browser Video Decode: Enabled
Browser Video Encode: Disabled
Browser Vulkan: Disabled
Browser WebGL: Enabled
Browser WebGL2: Enabled
Browser WebGPU: Disabled
Browser WebNN: Disabled
Video Card:
Driver: AMD AMD Radeon RX 580 Series (radeonsi, polaris10, LLVM 18.1.8, DRM 3.59, 6.11.4-zen1-1-zen)
Driver Version: 4.6 (Compatibility Profile) Mesa 24.2.5-arch1.1

**Comment by ConnorJC3** [comment]
Same here - unsetting `SDL_VIDEODRIVER` works (and I can then launch the beta), but I am setting `SDL_VIDEODRIVER` to `wayland` _on purpose_ because this is a wayland environment.

Thus steam should probably manually override/ignore this env var if it isn't going to support wayland (or even better, steam could start supporting wayland - that would be one more application off the X-only list).

**Comment by smcv** [comment]
> I am setting `SDL_VIDEODRIVER` to `wayland` _on purpose_ because this is a wayland environment

Please be aware that this is not necessarily an appropriate line of reasoning: even in a Wayland environment, and even outside Steam, some aspects of SDL (and especially SDL 2) work better if they are going via its X11 video driver, which in your case will talk to Xwayland.

If the maintainers of SDL considered its Wayland video driver to be ready to be used by default in Wayland environments, they would already have made it the default. The fact that you have needed to force it by setting an environment variable is a sign that they do not consider it to be ready for that.

**Comment by smcv** [comment]
@kisak-valve or @ConnorJC3, please could you retitle this to something like

`"Steamwebhelper is not responding" in latest Steam beta with SDL_VIDEODRIVER=wayland set`

to separate this from other reasons why the same symptom can appear? That will help to keep it on-topic and keep the signal-to-noise ratio high, which makes it more likely that it can be fixed.

**Comment by smcv** [comment]
If you do want to force SDL games to default to Wayland, overruling the SDL maintainers' recommendation, then a more stable way to achieve that would be to export `SDL_VIDEODRIVER=wayland,x11`, which means: try to use the Wayland video driver, but if that isn't possible, fall back to the X11 video driver. I suspect you might find that this works as a solution or workaround for this issue.

This is safer to use that than `SDL_VIDEODRIVER=wayland`, because not every build of SDL has the Wayland driver enabled: even if the copy of SDL from your operating system does, other copies of SDL included with Steam or with individual games do not necessarily have that, especially if they were compiled against very old library stacks for maximum compatibility between different OSs.

**Comment by smcv** [comment]
This should be fixed by Steam client beta 1729917543 (2024-10-26). I no longer see the reported issue on a system where I was previously able to reproduce it (AMD/Mesa + Arch Linux + KDE Plasma + Wayland).

Steam client beta 1729917543 (2024-10-26) automatically converts `SDL_VIDEODRIVER=wayland` into `SDL_VIDEODRIVER=wayland,x11`, with this warning logged:

```
warning: SDL_VIDEODRIVER='wayland' does not allow fallback, use 'wayland,x11' instead
```

It also does the same for `SDL_VIDEO_DRIVER` (used by SDL 3).

The practical result is that any SDL game that can use native Wayland will still use native Wayland, but games and components that cannot use native Wayland for whatever reason will automatically fall back to X11 via Xwayland.

This doesn't fix any other Wayland-related issues, notably #8020, but it also doesn't make them any worse.

**Comment by kisak-valve** [comment]
Closing per the last comment.
